### PR TITLE
arp-74440 add ARP to USiP config

### PR DESCRIPTION
**Pull request by gabezurita**
## Summary
Since ARP will use the sign-in service, we must add our app to the Unified Sign-In Service. To do this, we need to add a small amount of configuration code to several files detailed [in the documentation](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/tree/master/teams/vsp/teams/Identity/Product%20Documentation/Unified%20Sign%20in%20Page#-adding-an-external-app-to-usip-vets-website).

## Related issue(s)
[Zenhub issue](https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/74440)

## Acceptance criteria
The ARP app has been added to the Unified Sign-In Page config

### CODE REVIEW

File: src/platform/user/authentication/config/staging.config.js [hunk 1]
```diff
@@ -85,4 +85,17 @@ export default {
     oAuthOptions: defaultMobileOAuthOptions,
     externalRedirectUrl: EXTERNAL_REDIRECTS[EXTERNAL_APPS.VA_OCC_MOBILE],
   },
+  [EXTERNAL_APPS.ARP]: {
+    allowedSignInProviders: { ...defaultSignInProviders }, // TODO: Confirm we want all default sign-in providers
```
Comment (amprokop) [inline src/platform/user/authentication/config/staging.config.js:89]: Might just be ID.me and Login.gov?

File: src/platform/user/authentication/constants.js [hunk 2]
```diff
@@ -110,6 +111,7 @@ export const EXTERNAL_REDIRECTS = {
   [EXTERNAL_APPS.EBENEFITS]: `${eAuthURL}/ebenefits`,
   [EXTERNAL_APPS.VA_FLAGSHIP_MOBILE]: '',
   [EXTERNAL_APPS.VA_OCC_MOBILE]: `${eAuthURL}/MAP/users/v2/landing`,
+  [EXTERNAL_APPS.BENEFTS_ARP]: `/representative`, // TODO: Confirm this is correct
```
Comment (amprokop) [inline src/platform/user/authentication/constants.js:114]: Typo: benefits

**Comment by nihil2501** [comment]
It's not in the [docs for USiP config](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/tree/master/teams/vsp/teams/Identity/Product%20Documentation/Unified%20Sign%20in%20Page), but another configurational change is needed [here](https://github.com/department-of-veterans-affairs/vets-website/blob/1dcc7bb71705a323c7bfded2d1168745f3d0d8f7/src/platform/user/authentication/utilities.js#L97) to actually store in session the correct return url for a given external application. This bit of code allow lists and massages values on top of the config from the docs for some reason. We can remember to ask frontend identity about this at our sync if we're curious.

File: src/platform/user/authentication/config/staging.config.js (hunk 1 above)
Comment (nihil2501) [reply src/platform/user/authentication/config/staging.config.js:89]: @amprokop In other words, while [VA.gov](VA.gov) behavior is:
```
{
  allowedSignInProviders: {
    logingov: true,
    idme: true,
    dslogon: true,
    mhv: true,
  },
  allowedSignUpProviders: {
    logingov: true,
    idme: true,
  },
}
```
We will actually prefer to be more restrictive and exclude existing `dslogon` and `mhv` accounts from signing in?

**Review by asg5704 — DISMISSED** [review]

**Review by asg5704 — APPROVED** [review]
Tested and approved

**Review by micahchiang — APPROVED** [review]

File: src/platform/user/authentication/config/staging.config.js (hunk 1 above)
Comment (raudabaugh) [reply src/platform/user/authentication/config/staging.config.js]: Hi @nihil2501 - yes, we want to exclude existing `dslogon` + `mhv` (this was confirmed in an enablement team meeting today)

**Pull request merged**